src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -471,14 +494,134 @@ void TableDockWidget::addRow(PeakGroup* group, QTreeWidgetItem* root) {
+     * to show already bookmarked group with same mz and rt value.
+     * If user press <save> button <addSameMzRtGroup> sets to true which will be used
+     * to add this group's corresponding compound name to already bookmarked group
+     * of same rt and mz value.
+    */
+    addSameMzRtGroup=true;
+    promptDialog->close();
+}
+
+void TableDockWidget::rejectGroup(){
+    /**
+     * @details-this is a slot tied with <save> button of prompt dialog <promptDialog>
+     * to show already bookmarked group with same mz and rt value.
+     * If user press <cancel> button <addSameMzRtGroup> sets to false which will be used
+     * to reject this group's corresponding compound name.
+    */
+    addSameMzRtGroup=false;
+    promptDialog->close();
+}
+
+void ListView::keyPressEvent(QKeyEvent * event){
+    /**
+     * @details- this method will execute when user select and press <ctrl + c> to copy
+     * list of compound from prompt dialog which show already bookmarked group
+     * with same rt and mz value.
+    */
+    if (event->matches(QKeySequence::Copy)) {
+        /**set all selected compound name to clipboard*/
+        QApplication::clipboard()->setText(strings.join("\n"));
+      }
+}
+
+void TableDockWidget::showSameGroup(int sameMzRtGroupIndexHash){
+    /**
+     * @details- this method will set the prompt dialog to show already bookmared
+     * groups.
+    */
+
+    QStringList list;
+
+    for(int i=0;i<sameMzRtGroups[sameMzRtGroupIndexHash].size();++i){
+        /**saving all compound name of same rt and mz value to <list> variable*/
+        list.append(sameMzRtGroups[sameMzRtGroupIndexHash][i]);
+        qDebug()<<sameMzRtGroups[sameMzRtGroupIndexHash][i];
+    }
+
+    stringModel->setStringList(list);
+
+    listTextView->setModel(stringModel);
+    listTextView->setData(list);
+    listTextView->setSelectionMode(QAbstractItemView::MultiSelection);
+    listTextView->setEditTriggers(QAbstractItemView::NoEditTriggers);
+
+    /**set all widget and labels to prompt dailog*/
+    promptDialogLayout->insertWidget(0,upperLabel);
+    promptDialogLayout->insertWidget(1,listTextView);
+    promptDialogLayout->insertWidget(2,lowerLabel);
+    promptDialogLayout->insertLayout(3,buttonLayout);
+    promptDialog->setLayout(promptDialogLayout);
+    promptDialog->exec();
+
+}
+
 bool TableDockWidget::hasPeakGroup(PeakGroup* group) {
+
+    /**
+     * @detail- this function return true if this group already present in
+     * bookmrked group <allgroups>.
+     * In case of groups with same mz and rt value, this will holds its execution
+     * by QEventLoop <loop> and show a prompt to user whether he wants to add
+     * this group to bookmarked groups or not by executing method <showSameGroup>.
+     * <loop> will hold execution of this funtion till user press a button on prompt
+     * dialog.
+    */
+
+
+    int intMz=std::ceil(group->meanMz);
+    int intRt=std::ceil(group->meanRt);
+    int sameMzRtGroupIndexHash=intMz*intRt;
```

Comment:
Mz and Rt product can be same for multiple combinations of Mz and Rt. Try another approach maybe.

Reply:
@shubhra-agrawal QPair is now used as index.